src/core/qgstiles.h

```diff
@@ -106,6 +106,12 @@ class CORE_EXPORT QgsTileMatrix
     //! Returns a tile matrix for the usual web mercator
     static QgsTileMatrix fromWebMercator( int mZoomLevel );
 
+    //! Returns a tile matrix for a specific CRS
+    static QgsTileMatrix fromCustomDef( int mZoomLevel, const QString &crsAuthId, const QgsPointXY &z0TopLeftPoint, const double &scaleDenom0 );
```

Comment:
Can you change this to:

```suggestion
    static QgsTileMatrix fromCustomDef( int mZoomLevel, const QgsCoordinateReferenceSystem &crs, const QgsPointXY &z0TopLeftPoint, const double &scaleDenom0 );
```

Not all CRS objects have an auth id